assets/js/common/AlertingSettingsConfig/AlertingSettingsConfig.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import { noop } from 'lodash';
+
+import Button from '@common/Button';
+import DisabledGuard from '@common/DisabledGuard';
+import Tooltip from '@common/Tooltip';
+
+const alertingSettingsPermittedFor = ['all:alerting_settings'];
+
+export const ENFORCED_FROM_ENV_MESSAGE =
+  'Alerting settings are enforced by environment variables';
+
+export default function AlertingSettingsConfig({
+  settings: {
+    alertingEnabled = false,
+    smtpServer = 'https://.....',
+    smtpPort = 587,
+    smtpUsername = '.....',
+    senderEmail = '...@...',
+    recipientEmail = '...@...',
+    enforcedFromEnv = false,
+  } = {},
+  userAbilities = [],
```

Comment:
Not for this particular case but keep in mind this https://github.com/facebook/react/issues/18123#issuecomment-591978544

TL;DR: if an array prop is used inside a `useEffect` we'll have to define the empty array as a const